### PR TITLE
DROOLS-4936 : Unsaved changes dialog appears even if it just have been saved

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.redhat.com/browse/DROOLS-4936

### CODE REVIEW

**Comment by Rikkola** [comment]
Not sure if a test makes sense. Since after and before the test the results are the same. Unless the test is ran in GWT, but GWTUnit tests are too slow to use.

**Comment by jomarko** [comment]
@Rikkola I think we need to add unit tests, at least to confirm null values of fields that are used for hashcode computation doesn't cause an exception Thank you.

**Comment by Rikkola** [comment]
@jomarko I added some tests, however with or without the change in the main code they pass.

**Comment by jomarko** [comment]
@Rikkola then they do not test your change.

**Comment by Rikkola** [comment]
@jomarko Yes that is the problem. Only way to test the change is to use GWTTestCase, but we should not use them since they are slow.

**Comment by Rikkola** [comment]
Jenkins please retest this.

File: drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Test
+    public void defaultValues() {
+        final CEPWindow cepWindow = new CEPWindow();
+
+        assertNotNull(cepWindow.hashCode());
+    }
+
+    @Test
+    public void checkNulls() {
```
Comment (dupliaka) [inline drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java:114]: What about case if only parameters or only operator is set to null?

Comment (Rikkola) [reply drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java:114]: I added more test coverage for these situations.

**Review by dupliaka — APPROVED** [review]
Tested manually. 
Thanks for fix!

**Comment by Rikkola** [comment]
@manstis Could you also take a look at this PR please.

File: drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package org.drools.workbench.models.datamodel.rule;
```
Comment (manstis) [inline drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java:16]: License please.

File: drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java (hunk 1 above)
Comment (manstis) [reply drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java:114]: IDK if it's an overkill but (at least) one test creating two equal instances of `CEPWindow` and verifying `hashCode`s are equal may be nice....

**Comment by manstis** [comment]
Otherwise works fine (given the test on the RHPAM JIRA).

File: drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java (hunk 1 above)
Comment (Rikkola) [reply drools-workbench-models/drools-workbench-models-datamodel-api/src/test/java/org/drools/workbench/models/datamodel/rule/CEPWindowTest.java:114]: @manstis I improved this test to not only check the hashcode, but to compare hashcodes between other instances. I hope the other workbench objects do not get jealous :)

**Review by manstis — APPROVED** [review]

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I have a merge please.
